src/test/java/org/jenkinsci/plugins/workflow/cps/nodes/StepNodeTest.java

```diff
@@ -78,4 +83,87 @@
         r.assertLogContains("[Pipeline] // configFileProvider", b);
     }
 
+    @Test public void metastepConsoleShellClass() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(
+            "node {\n" +
+            "  wrap([$class: 'ConfigFileBuildWrapper', managedFiles: []]) {\n" +
+            "    writeFile text: '''<testsuite name='a'><testcase name='c'><error>failed</error></testcase></testsuite>''', file: 'x.xml'\n" +
+            "    step([$class: 'JUnitResultArchiver', testResults: 'x.xml'])\n" +
+            "  }\n" +
+            "}", true));
+        WorkflowRun b = r.assertBuildStatus(Result.UNSTABLE, p.scheduleBuild2(0));
+        List<FlowNode> coreStepNodes = new DepthFirstScanner().filteredNodes(b.getExecution(), new NodeStepTypePredicate("step"));
+        assertThat(coreStepNodes, hasSize(1));
+        assertEquals("junit", coreStepNodes.get(0).getDisplayFunctionName());
```

Comment:
We may as well show the nice view even if you have not yet updated your script to take advantage of simplified syntax.